pkg/lib/operatorclient/service.go

```diff
@@ -28,11 +28,22 @@ func (c *Client) DeleteService(namespace, name string, options *metav1.DeleteOpt
 // UpdateService will update the given Service resource.
 func (c *Client) UpdateService(service *v1.Service) (*v1.Service, error) {
 	klog.V(4).Infof("[UPDATE Service]: %s", service.GetName())
-	oldSa, err := c.GetService(service.GetNamespace(), service.GetName())
+	old, err := c.GetService(service.GetNamespace(), service.GetName())
 	if err != nil {
 		return nil, err
 	}
-	patchBytes, err := createPatch(oldSa, service)
+	normalized, err := cloneAndNormalizeObject(old)
+	if err != nil {
+		return nil, fmt.Errorf("failed to normalize existing Service resource for patch: %w", err)
+	}
+	if service.Spec.ClusterIP == old.Spec.ClusterIP {
+		// Support updating to manifests that specify a
+		// ClusterIP when its value is the same as that of the
+		// existing Service.
+		service = service.DeepCopy()
+		service.Spec.ClusterIP = ""
```

Comment:
A little confused -- the objective is to ensure the `ClusterIP` remains the same for the new service as the old service right? Even if other things about the service changed. 

`cloneAndNormalizeObject` strips the `ClusterIP` from the old service. This line removes the `ClusterIP` from the new service. This way, when we create the patch, we get a Service with no `ClusterIP` set. Then when we go to patch the existing service on the cluster with the patch, it preserves the existing `ClusterIP` of the service on cluster? But OLM can potentially update other Service fields. Makes sense.

Reply:
That's right -- my intention was to limit this to only allow thetwo cases I mentioned in my other [comment](https://github.com/operator-framework/operator-lifecycle-manager/pull/1884#discussion_r534483523).